BedrockServer.cpp

```diff
@@ -2209,20 +2035,268 @@ void BedrockServer::_finishPeerCommand(unique_ptr<BedrockCommand>& command) {
 void BedrockServer::_acceptSockets() {
     Socket* s = nullptr;
     Port* acceptPort = nullptr;
-    while ((s = acceptSocket(acceptPort))) {
+    while ((s = acceptUnlistedSocket(acceptPort))) {
         if (SContains(_portPluginMap, acceptPort)) {
             BedrockPlugin* plugin = _portPluginMap[acceptPort];
             // Allow the plugin to process this
             SINFO("Plugin '" << plugin->getName() << "' accepted a socket from '" << s->addr << "'");
             plugin->onPortAccept(s);
 
             // Remember that this socket is owned by this plugin.
-            SASSERT(!s->data);
             s->data = plugin;
         }
     }
 }
 
+STCPManager::Socket* BedrockServer::acceptUnlistedSocket(STCPServer::Port*& portOut) {
+    // Initialize to 0 in case we don't accept anything. Note that this *does* overwrite the passed-in pointer.
+    portOut = 0;
+    Socket* socket = nullptr;
+
+    // See if we can accept on any port
+    lock_guard <decltype(portListMutex)> lock(portListMutex);
+    for (Port& port : portList) {
+        // Try to accept on the port and wrap in a socket
+        sockaddr_in addr;
+        int s = S_accept(port.s, addr, true); // Note that this is set to be blocking.
```

Comment:
If this is blocking (according to the comment) won't it delay completing for any instance with a minimally active port? (e.g. it will accept the first new socket for 4444 and spin out a thread appropriately, but when it checks the admin port afterward, it will just block?)

Reply:
I think you're right. Will re-think.

Reply:
No, this is OK, just the comment is misleading. Updating.